MemoryMonitor: use /proc/*/oom_score_adj rather than /proc/*/oom_adj

morphis opened the pull request:
On newer kernels /proc/_/oom_adj is deprecated and should not be used anymore. As Open
webOS requires at least a linux 3.3 kernel this changes the use to the new
/proc/_/oom_score_adj entry.

Running a 3.0.49 kernel together with LunaSysMgr shows the following in kernel log:

[   28.763610] WebAppMgr (2090): /proc/2090/oom_adj is deprecated, please use
/proc/2090/oom_score_adj instead.

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch morphis@gravedo.de

morphis committed:
MemoryMonitor: use /proc/*/oom_score_adj rather than /proc/*/oom_adj

On newer kernels /proc/*/oom_adj is deprecated and should not be used anymore. As Open
webOS requires at least a linux 3.3 kernel this changes the use to the new
/proc/*/oom_score_adj entry but still keeps backward compatiblity for older kernels which
onles provides /proc/*/oom_adj.

More information about the change are available on the following pages:
- https://github.com/mirrors/linux-2.6/commit/a63d83f427fbce97a6cea0db2e64b0eb8435cd10#include/linux/oom.h
- https://www.redhat.com/archives/lvm-devel/2011-July/msg00097.html

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>

ericblade-owo commented:
If I'm not mistaken, the numbers that oom_score_adj uses are intended to be different from what oom_adj used, so I'm thinking that setting, as the code right below this change does, -17 as the oom_score_adj has a rather different effect than setting -17 as the oom_adj did.

Also, I'd advise for checking the existence of oom_score_adj and using that, otherwise using oom_adj, considering that the primary platforms supported for Luna on desktop are Ubuntu 11 and 12, which use kernels all the way back to 2.6.38.

morphis commented:
You're right. oom_score_adj has a different use of the supplied values. It defines a valid range fro -1000 up to +1000 (see https://github.com/mirrors/linux-2.6/commit/a63d83f427fbce97a6cea0db2e64b0eb8435cd10#include/linux/oom.h for more details). Will adjust the patch accordingly.

As far as I know the relevant patch should be included in 2.6.38 already so we don't have to check for the old entry.

morphis commented:
I updated the commit to include support for /proc/_/oom_adj and /proc/_/oom_score_adj. For both implementations the correct values are set (-17 for oom_adj and -1000 for oom_score_adj). Furthermore I removed the change related to WebAppManager. I will issue a pull-request with the same change for this separated component in the next days.

morphis commented:
Reworked the code again. Code relevant to adjusting OOM score is now completely separted from the ctor. Code is tested on my Galaxy Nexus.

RS102839 commented:
cherry-picked
